PlainTasksDates.py

```diff
@@ -548,19 +548,19 @@ def extract_tag(self, point):
             tag under cursor (i.e. point)
         '''
         start = end = point
-        tag_pattern = r'(?<=\s)(\@[^\(\) ,\.]+)([\w\d\.\(\)\-!? :\+]*)'
-        line = self.view.line(point)
-        matches = re.finditer(tag_pattern, self.view.substr(line))
-        for match in matches:
-            m_start = line.a + match.start(1)
-            m_end   = line.a + match.end(2)
-            if m_start <= point <= m_end:
-                start = line.a + match.start(2)
-                end   = m_end
-                break
-        else:
-            match = None
-        tag = match.group(0) if match else ''
+        tag = ''
+        if 'tag.todo' in self.view.scope_name(point):
+            reg = self.view.extract_scope(point)
```

Comment:
This is not reliable enough, e.g. there is option to show date preview for due tag within paranthesis `"due_preview_offset": 1`, as I mentioned in prev pr phantom will change result of `extract_scope`.
We also should always expect that view may be affected by other plugins so even if user has default `"due_preview_offset": 0`, still there might be other phantoms.

I think we could try to use `view.find` method, i.e. check if cursor is on tag then search `@` which is related to this tag (it will be either somewhere on left side of cursor or it might be exactly on right side of cursor), then the point of `@` is a starting position for `view.find` which allow to use the regex from syntax, i.e. with atomic group etc. Right?